sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_client_async.py

```diff
@@ -483,13 +480,14 @@ async def get_blob_properties(self, **kwargs):
                 snapshot=self.snapshot,
                 lease_access_conditions=access_conditions,
                 modified_access_conditions=mod_conditions,
-                cls=deserialize_blob_properties,
+                cls=kwargs.pop('cls', None) or deserialize_blob_properties,
                 cpk_info=cpk_info,
                 **kwargs)
         except StorageErrorException as error:
             process_storage_error(error)
         blob_props.name = self.blob_name
-        blob_props.container = self.container_name
+        if isinstance(blob_props, BlobProperties):
```

Comment:
same here 

https://docs.python.org/2/glossary.html#term-eafp

interesting read about why - Easier to ask for forgiveness than permission